packages/server/src/workers/subscription.ts

```diff
@@ -446,12 +449,20 @@ export async function execSubscriptionJob(job: Job<SubscriptionJobData>): Promis
   }
 }
 
-async function tryGetSubscription(systemRepo: Repository, subscriptionId: string): Promise<Subscription | undefined> {
+async function tryGetSubscription(
+  systemRepo: Repository,
+  subscriptionId: string,
+  channelType: SubscriptionJobData['channelType'] | undefined
+): Promise<Subscription | undefined> {
   try {
-    return await systemRepo.readResource<Subscription>('Subscription', subscriptionId);
+    return await systemRepo.readResource<Subscription>('Subscription', subscriptionId, {
+      checkCacheOnly: channelType === 'websocket',
+    });
   } catch (err) {
     const outcome = normalizeOperationOutcome(err);
-    if (isGone(outcome)) {
+    // If the Subscription was marked as deleted in the database, this will return "gone"
+    // However, deleted WebSocket subscriptions will return "not found"
+    if (isGone(outcome) || isNotFound(outcome)) {
```

Comment:
good catch 👍 

This would have also applied to "purged" or "expunged" subscriptions, which are hard deleted from the database.